.github/workflows/ci.yml

```
@@ -16,66 +23,78 @@ jobs:
-            version: 11
-          - distro: temurin
-            version: 17
+        scala: [2.12.17, 2.13.10, 3.2.0]
+        java: [temurin@11, temurin@17]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout current branch (full)
-        uses: actions/checkout@v3
+        uses: actions/checkout@v2
         with:
           fetch-depth: 0
 
-      - name: Setup Java
-        uses: actions/setup-java@v3
+      - name: Download Java (temurin@11)
+        id: download-java-temurin-11
+        if: matrix.java == 'temurin@11'
+        uses: typelevel/download-java@v1
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Setup Java (temurin@11)
+        if: matrix.java == 'temurin@11'
+        uses: actions/setup-java@v2
+        with:
+          distribution: jdkfile
+          java-version: 11
+          jdkFile: ${{ steps.download-java-temurin-11.outputs.jdkFile }}
+
+      - name: Download Java (temurin@17)
+        id: download-java-temurin-17
+        if: matrix.java == 'temurin@17'
+        uses: typelevel/download-java@v1
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Java (temurin@17)
+        if: matrix.java == 'temurin@17'
+        uses: actions/setup-java@v2
+        with:
+          distribution: jdkfile
+          java-version: 17
+          jdkFile: ${{ steps.download-java-temurin-17.outputs.jdkFile }}
+
+      - name: Cache sbt
+        uses: actions/cache@v2
         with:
-          distribution: ${{ matrix.java.distro }}
-          java-version: ${{ matrix.java.version }}
-          cache: sbt
+          path: |
+            ~/.sbt
+            ~/.ivy2/cache
+            ~/.coursier/cache/v1
+            ~/.cache/coursier/v1
+            ~/AppData/Local/Coursier/Cache/v1
+            ~/Library/Caches/Coursier/v1
+          key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Check that workflows are up to date
+        run: sbt githubWorkflowCheck
 
       - name: Build project
         run: sbt test
 
+      - name: Make target directories
+        if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/v'))
+        run: mkdir -p core/target/1-15-jvm-2.13 core/target/1-js-3 core/target/1-js-2.13 joda/target/1-13-jvm-2.12 core/target/1-js-2.12 core/target/1-13-jvm-2.12 core/target/1-15-jvm-2.12 core/target/1-15-js-2.12 joda/target/1-15-jvm-3 joda/target/1-15-jvm-2.13 core/target/1-15-js-2.13 core/target/1-jvm-2.12 joda/target/1-14-jvm-2.13 core/target/1-native-3 target core/target/1-15-jvm-3 core/target/1-15-js-3 joda/target/1-14-jvm-2.12 core/target/1-15-native-3 joda/target/1-jvm-2.13 joda/target/1-15-jvm-2.12 core/target/1-14-jvm-2.12 joda/target/1-jvm-2.12 core/target/1-14-jvm-2.13 core/target/1-jvm-2.13 core/target/1-jvm-3 joda/target/1-jvm-3 core/target/1-15-native-2.13 core/target/1-native-2.13 project/target
```

I'm guessing that extra directories would not kill this.  If so, you could simplify this greatly:

```suggestion
        run: mkdir -p {core,joda}/target/1-{13,14,15,}-{js,jvm,native}-{2.12,2.13,3} {project/,}target
```